pygradle-plugin/src/test/groovy/com/linkedin/gradle/python/tasks/action/PipActionHelpers.groovy

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.tasks.action.pip
+
+import com.linkedin.gradle.python.util.DefaultPackageSettings
+import com.linkedin.gradle.python.util.PackageInfo
+import org.junit.rules.TemporaryFolder
+
+import java.nio.file.Paths
+
+class PipActionHelpers {
+    private PipActionHelpers() {
+        //NOOP
+    }
+
+    static class RequiresRebuildOverridePackageSettings extends DefaultPackageSettings {
+        private final List<String> override
+
+        RequiresRebuildOverridePackageSettings(TemporaryFolder temporaryFolder, List<String> override) {
+            super(temporaryFolder.root)
+            this.override = override
+        }
+
+        @Override
+        boolean requiresSourceBuild(PackageInfo packageInfo) {
+            return override.contains(packageInfo.name)
```

Comment:
To match the deleted integration tests and semantic intent, this methods and other methods in this test class should call super when not matching the package name. Otherwise, any future change in the default class will **not** match these tests. I believe this should be `return (override.contains(packageInfo.name)) ? true : super.requiresSourceBuild(packageInfo)`. Similarly for others.

Reply:
I don't think that's needed. We are trying to force the behavior for the tests, so we don't really care if these tests don't do what the super class does.

If the behavior of the default class changes, then the tests should be updated there. These tests should be isolate from changes to DefaultPackageSettings.